osu.Game/Rulesets/Difficulty/Skills/PreStrainSkill.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using osu.Game.Rulesets.Difficulty.Preprocessing;
+using osu.Game.Rulesets.Mods;
+
+namespace osu.Game.Rulesets.Difficulty.Skills
+{
+    public abstract class PreStrainSkill : StrainSkill
+    {
+        protected abstract double StrainDecayBase { get; }
+
+        protected abstract double SkillMultiplier { get; }
```

Comment:
nothing ever actually reads this value (same goes for `GetAllStrainPeaks()`)

Reply:
future feature but okay
after this the class role is not for strain

Reply:
for example using strain
https://github.com/GoldenMine0502/osu/blob/tapvariance/osu.Game.Rulesets.Osu/Difficulty/Skills/Pre/NoteVarianceAngle.cs